pkg/osquery/runtime/runtime_helpers.go

```diff
@@ -33,3 +35,32 @@ func platformArgs() []string {
 func isExitOk(err error) bool {
 	return false
 }
+
+func ensureProperPermissions(o *OsqueryInstance, paths *osqueryFilePaths) error {
+	// The extensions file should be owned by the process's UID or the file
+	// should be owned by root. Osquery will refuse to load the extension
+	// otherwise
+	fd, err := os.Stat(paths.extensionPath)
```

Comment:
This isn't exported, will this work? (If we're only chowning the extension, I'd just pass that directly)

Reply:
It works because they're in the same package, but you're right about just passing the path directly, I'll do that.

Reply:
If you want to get fancy, you can use a signature like `func ensure(filepaths ...string)`